### PR TITLE
Feature/7859 make alt editable

### LINKED ISSUE #7859: Media Library: make the ALT field editable
This is a follow up to https://github.com/wordpress-mobile/WordPress-iOS/issues/7837.

Now that Media objects synchronize the `alt` field with our backend, we should make this field editable from the media library at: `My Sites` > `Pick a site` > `Media`.

### PR DESCRIPTION
**Pull request by 3vangelos**
Fixes #7859 

To test: - Is alt text editable in images.
              - make sure you cannot edit alt text in videos

Needs review: @diegoreymendez

### CODE REVIEW

**Comment by diegoreymendez** [comment]
I'm adding @frosty as our second reviewer.

**Comment by 3vangelos** [comment]
@nheagy sorry to ping you, but it seems like this PR also did not get picked up by Buddy build. Can I do anything so my PRs get pulled up automatically or will I have to ping somebody? 🙇

**Comment by diegoreymendez** [comment]
@3vangelos - Don't worry too much about it.  Once the reviews are complete we can manually make BB build this branch.

File: WordPress/Classes/ViewRelated/Media/MediaItemViewController.swift [hunk 1]
```diff
@@ -325,6 +341,17 @@ class MediaItemViewController: UITableViewController {
         }
     }
 
+    private func editAlt() -> ((ImmuTableRow) -> ()) {
+        return { [weak self] row in
+            let editableRow = row as! EditableTextRow
+            self?.pushSettingsController(for: editableRow, hint: NSLocalizedString("Image Alt", comment: "Hint for image description on image settings."),
```
[the remark this answers is not in the document]
Comment (3vangelos) [reply WordPress/Classes/ViewRelated/Media/MediaItemViewController.swift]: @frosty, out that is a bit embarrassing. It's fixed!

**Comment by diegoreymendez** [comment]
Buddybuild is failing but all is working fine on my end.  Merging.

**Comment by elibud** [comment]
@diegoreymendez please keep an eye on develop on buddybuild just to make sure the merge was good.

**Comment by diegoreymendez** [comment]
The problem that I see right now with BB (and some other tools) is that 99% of the failures they show are false positives.

I'll check `develop` out no problem, but it does seem to me like our tools are adding a lot of meta-work for little advantage right now?

**Comment by diegoreymendez** [comment]
For clarity: I just checked the latest from `develop` and it's both building fine, and the unit tests are not failing.

**Comment by elibud** [comment]
I haven't really seen false negatives from buddybuild or Hound, but I am not happy with how long buddybuild takes to build, on release days it's a nightmare. If you see any false negative we should fix them as they appear, @koke can advise us about what the best course of action is for each issue.

**Comment by diegoreymendez** [comment]
There's a false negative right in this PR.  Failing tests not failing for the reviewers, that is.

**Comment by elibud** [comment]
Did you check it on buddybuild? What was it saying? For some reason I can't find the info about that branch in buddybuild's page

**Comment by diegoreymendez** [comment]
A nil date test was failing and there are also some CoreData errors which I have no insights about.

Here are the logs in case you want to investigate further:
https://dashboard.buddybuild.com/apps/57a120bbe0f5520100e11c19/build/59d25d90fbce0400019b7d42#logs

**Comment by diegoreymendez** [comment]
Just for history: this wasn't a false positive.  The tests I was checking were not really failing tests... they were just raising expected exceptions.

The tests in this PR were failing due to a randomly appearing bug introduced with Helpshift 6.2.0.

**Pull request merged**